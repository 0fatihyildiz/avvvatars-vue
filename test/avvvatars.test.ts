import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import AvvvatarsVue from '../src/components/Avvvatars.vue'

const value = 'Avvvatars'

describe('Avvvatars Snapshots', () => {
  test('value', () => {
    const wrapper = mount(AvvvatarsVue, {
      props: {
        value,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('displayValue', () => {
    const wrapper = mount(AvvvatarsVue, {
      props: {
        displayValue: value,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Value & displayValue', () => {
    const wrapper = mount(AvvvatarsVue, {
      props: {
        value,
        displayValue: 'test',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Size', () => {
    const wrapper = mount(AvvvatarsVue, {
      props: {
        value,
        size: 50,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Shape', () => {
    const wrapper = mount(AvvvatarsVue, {
      props: {
        value,
        variant: 'shape',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Shadow & Border', () => {
    const wrapper = mount(AvvvatarsVue, {
      props: {
        value,
        shadow: true,
        border: true,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
