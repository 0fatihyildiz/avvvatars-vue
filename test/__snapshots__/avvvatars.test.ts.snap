// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avvvatars Snapshots > Shadow & Border 1`] = `
"<div data-v-d64f069a="" class="avvvatars" style="border: 2px solid #fff; box-shadow: 0px 3px 8px rgba(18, 18, 18, 0.04),
        0px 1px 1px rgba(18, 18, 18, 0.02);">
  <p data-v-d64f069a="" class="text">Av</p>
</div>"
`;

exports[`Avvvatars Snapshots > Shape 1`] = `"<div data-v-d64f069a="" class="avvvatars"><span data-v-d64f069a="" class="shape"><svg data-v-d64f069a="" width="16" height="16" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg" role="img"><path d="M16 0l3.856 9.321 10-1.321-6.144 8 6.145 8-10.001-1.321L16 32l-3.856-9.321-10 1.321 6.144-8-6.144-8 10 1.321L16 0z" fill="currentColor"></path></svg></span></div>"`;

exports[`Avvvatars Snapshots > Size 1`] = `
"<div data-v-d64f069a="" class="avvvatars">
  <p data-v-d64f069a="" class="text">Av</p>
</div>"
`;

exports[`Avvvatars Snapshots > Value & displayValue 1`] = `
"<div data-v-d64f069a="" class="avvvatars">
  <p data-v-d64f069a="" class="text">te</p>
</div>"
`;

exports[`Avvvatars Snapshots > displayValue 1`] = `
"<div data-v-d64f069a="" class="avvvatars">
  <p data-v-d64f069a="" class="text">Av</p>
</div>"
`;

exports[`Avvvatars Snapshots > value 1`] = `
"<div data-v-d64f069a="" class="avvvatars">
  <p data-v-d64f069a="" class="text">Av</p>
</div>"
`;
